### PR TITLE
Bug 1948719: update dependencies to 1.21

### PR DESCRIPTION
**Pull request by elmiko**
This change updates the go.mod and related vendor files to the
kubernetes 1.21 baseline. It also updates one import in the
`pkg/operator/operator.go` to allow vendoring. This was achieved by
 updating some of the entries in go.mod and then running `make vendor`.

The updates to go.mod were as follows:
```
github.com/openshift/api master
github.com/openshift/client-go master
github.com/openshift/library-go master
sigs.k8s.io/controller-runtime 0.9.0-alpha.1
```

Then ran `make vendor` to produce the update. During this process the
error in `operator.go` was discovered and fixed.

Also adds necessary changes to for controller-runtime updates.

### CODE REVIEW

**Comment by JoelSpeed** [comment]
@elmiko Seems there's a syntax error:
```
parse 0000_50_cluster-autoscaler-operator_03_rbac.yaml: error parsing: error converting YAML to JSON: yaml: line 7: did not find expected key",
```

**Comment by elmiko** [comment]
not sure why it failed there, but i found a syntax error in a different part of the file.

where did you see that failure @JoelSpeed ?

**Comment by JoelSpeed** [comment]
It was in the log of the upgrade test result

File: pkg/operator/webhookconfig.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (w *WebhookConfigUpdater) Start(stopCh <-chan struct{}) error {
 	klog.Infof("Webhook configuration status: %s", op)
 
 	// Block until the stop channel is closed.
-	<-stopCh
+	stop.Done()
```
Comment (Danil-Grigorev) [inline pkg/operator/webhookconfig.go:75]: This is not equivalent to previous code sample, should be 
```go
<-stop.Done()
```

File: pkg/operator/status.go [hunk 2]
```diff
@@ -309,10 +309,10 @@ func (r *StatusReporter) Start(stopCh <-chan struct{}) error {
 		return available, nil
 	}
 
-	err := wait.PollImmediateUntil(interval, pollFunc, stopCh)
+	err := wait.PollImmediateUntil(interval, pollFunc, stop.Done())
 
 	// Block until the stop channel is closed.
-	<-stopCh
+	stop.Done()
```
Comment (Danil-Grigorev) [inline pkg/operator/status.go:315]: This is not equivalent to previous code sample, should be 
```go
<-stop.Done()
```

**Review by Danil-Grigorev — COMMENTED** [review]
One minor change and it is lgtm

File: pkg/operator/webhookconfig.go (hunk 1 above)
Comment (elmiko) [reply pkg/operator/webhookconfig.go:75]: thanks Danil, i'll fix it up

**Comment by elmiko** [comment]
updated to include Danil's suggestion

**Pull request merged**